doc/source/reference/c-api/array.rst

```diff
@@ -790,25 +790,25 @@ Dealing with types
 General check of Python Type
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-.. c:function:: PyArray_Check(PyObject *op)
+.. c:macro:: PyArray_Check(op)
```

Comment:
Can you remind me why these need changing to `c:macro`?

Would this work instead as

```suggestion
.. c:function:: int PyArray_Check(PyObject *op)
```

Reply:
The reason is that `PyArray_Check` is defined as macro, so I don't think there's a problem with using `.. c:function::` here.

Reply:
```
#define PyArray_IsScalar(obj, cls) (PyObject_TypeCheck(obj, &Py##cls##ArrType_Type))
```
I don't know what type `cls` has. I think we have to use `.. c:macro:` for `PyArray_IsScalar`.

Reply:
Yes, `IsScalar` does have to be a macro. Where possible to ascribe types, we should do so and use `function`

Reply:
If we don't use `macro`, these changes don't depend on sphinx 3, and reduce some reference warnings. Should I separate these to another PR?

Reply:
Definitely, please do :)

Reply:
Done and merged in #17549!